### PR TITLE
Homework10: Ekaterina Zavadskaya

### PR DESCRIPTION
**Pull request by KateZavadskaya**
Add done homework10

Docs: hm10_1.py - hm10_3.py

### CODE REVIEW

File: hm10_1.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            if isinstance(y, int):
+                if y > 0:
+                    print(f"{i}. {y} Число больше 0")
+                elif y == 0:
+                    raise ValueError(f"{i}. {y} Число равно 0")
+                elif y < 0:
+                    raise ValueError(f"{i}. {y} Число меньше 0")
+            else:
+                raise ValueError(f"{i}. {y} Невозможно сравнить")
+                # i += 1
```
Comment (Stml89) [inline hm10_1.py]: remove it

File: hm10_1.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    """ Function takes arguments _any """
+
+    def wrapper(*args):
+
+        """ Function takes arguments in any value """
+
+        lst = []
+        for x in args:
+            lst.append(x)
```
Comment (Stml89) [inline hm10_1.py]: work with args, you do not need to copy these values to list()

File: hm10_1.py (hunk 1 above)
Comment (Stml89) [inline hm10_1.py]: Positive value, value less the 0

```suggestion
                if y < 0:
                    raise ValueError(f"{i}. {y} Невозможно сравнить")
```

File: hm10_2.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    """ Function takes arguments _r """
+
+    def wrapper(*args):
+
+        """ Function takes arguments in any value """
+
+        lst = []
+        for x in args:
+            lst.append(x)
```
Comment (Stml89) [inline hm10_2.py]: use `args`

File: hm10_2.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+@decorator
+def my_func(*args):
+
+    """ Function shows sum of arguments """
+
+    a = sum(args)
+    return a
+
+
+my_func(1, 9.1, "k")
```
Comment (Stml89) [inline hm10_2.py]: `assert ...`

File: hm10_3.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+def add(a, b):
+
+    """ Function work with str and int """
+
+    return a + str(b)
+
+
+assert add("9", 9) == "99", "Error processing (str, int)"
+assert add(9, 9) == "99", "Error processing (str, int)"
+assert add("9", "9") == "99", "Error processing (str, int)"
```
Comment (Stml89) [inline hm10_3.py:41]: what if the count of arguments is 3?

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: hm10_1.py (hunk 1 above)
Comment (KateZavadskaya) [reply hm10_1.py]: Deleted exrta code and used args, ok
Do not use your advice:
1. it is easier to understand the numeric of the value
2. we need to understand what we have: number or not. And only then compare

File: hm10_2.py (hunk 4 above)
Comment (KateZavadskaya) [reply hm10_2.py]: Used args, ok
I don’t understand how to use assert correctly, as it duplicates information. Please, help with the correct use with assert without duplicates information

File: hm10_3.py (hunk 5 above)
Comment (KateZavadskaya) [reply hm10_3.py:41]: Chanched arguments for 3 in first variant (not with #), ok.
But I understand, it also could be 5, 100, 100500. And I do not understand, how to fix it. 
If it needs 3 - done
If it needs 5, 100, 100500 - and as for me it should be so - I do not nderstand the correct fix, please help

File: hm10_3.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+            converted_args = []
+            for arg, arg_type in zip(args, arg_types):
+                converted_arg = arg_type(arg)
+                converted_args.append(converted_arg)
+            return func(*converted_args)
+        return wrapper
+    return decorator
+
+
+@typed(str, int, str)
```
Comment (Stml89) [inline hm10_3.py]: the requirements say that only 1 type must be passed
```suggestion
@typed(str) / @typed(int) / ....
```

File: hm10_3.py (hunk 5 above)
Comment (Stml89) [reply hm10_3.py:41]: Could you please rephrase your question , I'm not sure that understood it

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: hm10_1.py (hunk 1 above)
Comment (Stml89) [reply hm10_1.py]: Regarding `isinstance` agreed, but what about to change `if-elif-elif` to just 1 `if`?

Comment (KateZavadskaya) [reply hm10_1.py]: Mmm, the task is to lavidate the argument: is it positive. If not positive - tell it. 0 - not positive, not negative. If make "Positive value, value less the 0", what about 0 and not numeric/float.
Simple is better than complex, logical is better than confusing :) For me my logic is more logical :)

File: hm10_3.py (hunk 5 above)
Comment (KateZavadskaya) [reply hm10_3.py:41]: I do not understand is it difference between 3 arguments and indefinite quantity. 
Now I fix it for 3 arguments, but you can ask for 4, 7, 100 arguments.  In that cause, should I add the 4th agrument or my cod is incorrect? If it is incorrect - where is the mistake, I can not understand.

Comment (Stml89) [reply hm10_3.py:41]: In general, your solution will work with different counts of arguments. It uses `*args ` and`for`.
No need to write additional cases

**Review by Stml89 — APPROVED** [review]

**Pull request merged**